### PR TITLE
[Integration][SonarQube] Bug | 404 response when querying SonarQube issues

### PR DESCRIPTION
**Pull request by lordsarcastic**
# Description

What - Fixed a bug where SonarQube instances lower than 10.2 report a 404 error when retrieving issues

Why - This is because the issues list API on SonarQube was introduced in 10.2 and also as an internal API

How - Checking for the integration version and choosing the right API based on the integration version

## Type of change

Please leave one option from the following and delete the rest:

- [x] Bug fix (non-breaking change which fixes an issue)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots
image

## API Documentation

[SonarQube issues list API](https://next.sonarqube.com/sonarqube/web_api/api/issues/list?internal=true)

### CODE REVIEW

File: integrations/sonarqube/client.py [hunk 1]
```diff
@@ -338,12 +350,20 @@ async def get_issues_by_component(
         component_key = component.get("key")
         endpoint_path = ""
 
-        if self.is_onpremise:
+        current_version = await self._get_sonarqube_instance_version()
+
+        if current_version >= MINIMUM_SUPPORTED_ISSUES_LIST_VERSION:
+            logger.info(
+                f"SonarQube version is {current_version}. Using issues/list API"
```
Comment (PeyGis) [inline integrations/sonarqube/client.py:357]:
```suggestion
            logger.info(
                f"SonarQube version is {current_version}. Using {Endpoints.ISSUE_LIST} API"
```

File: integrations/sonarqube/client.py [hunk 2]
```diff
@@ -338,12 +350,20 @@ async def get_issues_by_component(
         component_key = component.get("key")
         endpoint_path = ""
 
-        if self.is_onpremise:
+        current_version = await self._get_sonarqube_instance_version()
+
+        if current_version >= MINIMUM_SUPPORTED_ISSUES_LIST_VERSION:
```
Comment (PeyGis) [inline integrations/sonarqube/client.py:355]: Can you verify if this logic will work for SonarCloud too? From the way I see it, they don't have issues/list, only issues/search

https://sonarcloud.io/web_api/api/issues/search?deprecated=false

**Review by PeyGis — CHANGES_REQUESTED** [review]
looks great. left some comments

File: integrations/sonarqube/client.py [hunk 3]
```diff
@@ -336,20 +337,13 @@ async def get_issues_by_component(
...
         component_key = component.get("key")
-        endpoint_path = ""
-
-        if self.is_onpremise:
-            query_params = {"project": component_key}
-            endpoint_path = Endpoints.ONPREM_ISSUES
-        else:
-            query_params = {"componentKeys": component_key}
-            endpoint_path = Endpoints.SAAS_ISSUES
+        query_params = {"componentKeys": component_key}
```
Comment (PeyGis) [inline integrations/sonarqube/client.py:340]:
```suggestion
        if self.is_onpremise:
            query_params = {"components": component_key}
        else:
            query_params = {"componentKeys": component_key}
```

File: integrations/sonarqube/client.py [hunk 4]
```diff
@@ -38,15 +38,16 @@ class Endpoints:
     WEBHOOKS = "webhooks"
     MEASURES = "measures/component"
     BRANCHES = "project_branches/list"
-    ONPREM_ISSUES = "issues/list"
-    SAAS_ISSUES = "issues/search"
+    ISSUES_SEARCH = "issues/search"
     ANALYSIS = "activity_feed/list"
     PORTFOLIO_DETAILS = "views/show"
     PORTFOLIOS = "views/list"
+    VERSION = "server/version"
```
Comment (PeyGis) [inline integrations/sonarqube/client.py:45]: delete since we don't need it

File: integrations/sonarqube/client.py [hunk 5]
```diff
@@ -38,15 +38,16 @@ class Endpoints:
...
 
 
 PAGE_SIZE = 100
 
+
```
Comment (PeyGis) [inline integrations/sonarqube/client.py:50]: remove blank line

**Review by PeyGis — CHANGES_REQUESTED** [review]
left some comments and suggestions

File: integrations/sonarqube/client.py (hunk 3 above)
Comment (lordsarcastic) [reply integrations/sonarqube/client.py:340]: Only components since we are using only one API

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**